gptq.py

```diff
@@ -183,38 +183,54 @@ def fasterquant(
-                d = Hinv1[i, i]
-                
-                if dct_mode == 1:
-
-                    if groupsize != -1:
-                        if not static_groups:
-                            if (i1 + i) % groupsize == 0:
-                                W_group = torch.split(W[:, (i1 + i):(i1 + i + groupsize)], groupsize, dim=1)
-                                dct_W_group = torch.stack([dct(column, norm='ortho') for column in W_group], dim=1)
-                                self.quantizer.find_params(dct_W_group, weight=True)
-                    else:
-                        idx = i1 + i
-                    if actorder:
-                        idx = perm[idx]
-                        self.quantizer = groups[idx // groupsize]
-
-                
-                    dct_w = dct(w.unsqueeze(0), norm='ortho').squeeze(0)
-                    quantized_dct_w = quantize(
-                        dct_w.unsqueeze(1), self.quantizer.scale, self.quantizer.zero, self.quantizer.maxq
-                    ).flatten()
-                    q = idct(quantized_dct_w.unsqueeze(0), norm='ortho').squeeze(0)
-                else:
-                    q = quantize(
-                        w.unsqueeze(1), self.quantizer.scale, self.quantizer.zero, self.quantizer.maxq
-                    ).flatten()
+            for i in range(count):  # loop over each column of W1
+                w = W1[:, i]  # get the current column
+                d = Hinv1[i, i]  # get the corresponding diagonal element of Hinv1
+
+                if dct_mode == 1:  # if DCT mode is enabled
+
+                    if groupsize != -1:  # if groupsize is not -1
+                        if not static_groups:  # if groups are not static
+                            if (i1 + i) % groupsize == 0:  # if the current column is the first column of a group
+                                W_group = torch.split(W[:, (i1 + i):(i1 + i + groupsize)], groupsize, dim=1)  # split W into groups
+                                dct_W_group = torch.stack([dct(column, norm='ortho') for column in W_group], dim=1)  # apply DCT to each group
+                                self.quantizer.find_params(dct_W_group, weight=True)  # find quantization parameters for each group
+                                dct_w = dct_W_group[:, i]  # get the DCT coefficients for the current column
+                                quantized_dct_w = quantize(
+                                    dct_w.unsqueeze(1), self.quantizer.scale, self.quantizer.zero, self.quantizer.maxq
+                                ).flatten()  # quantize the DCT coefficients
+                                q = idct(quantized_dct_w.unsqueeze(0), norm='ortho').squeeze(0)  # apply inverse DCT to get the quantized column
+                            else:  # if the current column is not the first column of a group
+                                dct_w = dct(w.unsqueeze(0), norm='ortho').squeeze(0)  # apply DCT to the current column
```

Comment:
This is code duplication. If you refactor your code into the logic it was in before, you will be able to avoid this.